src/patterns/ec2-app.test.ts

```diff
@@ -16,6 +17,64 @@ describe("the GuEC2App pattern", function () {
     expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
   });
 
+  it("adds the correct permissions for apps which need to fetch private config from s3", function () {
+    const stack = simpleGuStackForTesting();
+    const app = "test-gu-ec2-app";
+    new GuEc2App(stack, {
+      applicationPort: GuApplicationPorts.Node,
+      app: app,
+      publicFacing: false,
+      userData: {
+        distributable: {
+          bucket: GuDistributionBucketParameter.getInstance(stack),
+          fileName: "my-app.deb",
+          executionStatement: `dpkg -i /${app}/my-app.deb`,
+        },
+        configuration: {
+          bucket: new GuPrivateConfigBucketParameter(stack),
```

Comment:
for a different PR: should `GuPrivateConfigBucketParameter` be a singleton, similar to `GuDistributionBucketParameter`?

Reply:
Yes, I think so. Even if teams have private config split across multiple buckets it should be pretty trivial to copy & paste a couple of config files into a different bucket to avoid complexity in the template.